vllm/model_executor/layers/tuned_gemm.py

```diff
@@ -51,6 +51,31 @@ def create_ds(self):
     def query_sol(self, m, n, k):
         return self.solids.get((m, n, k), (0, 0))
 
+    def apply_skinny(self, _, n, k, inp_view, weights):
+        if n > 4 or k % 8 != 0 or inp_view.dtype != torch.float16:
+            return None
+        out = torch.empty(inp_view.shape[0],
+                            weights.shape[0],
+                            dtype=inp_view.dtype,
+                            device='cuda')
+        _custom_C.wvSpltK(weights, inp_view, out, n, self.cu_count)
```

Comment:
Not something that needs to be changed right now but we probably want to refactor this eventually so that the MP core count is done at the C++ level: IMO not good decomposition to have it here.